### PR TITLE
Use ansible-specdoc for documentation generation

### PR DESCRIPTION
**Pull request by LBGarber**
This branch adds the ability to dynamically generation documentation from module spec. This ensures there is a single source of truth for spec documentation.

The eventual goal will be to completely remove the need to inline documentation, but for the time being inline documentation will be automatically generated.

### CODE REVIEW

File: docs/modules/firewall.rst [hunk 1]
```diff
@@ -12,124 +12,143 @@ firewall
...
-      The unique ID of the device to attach to this Firewall.
 
+          addresses (False, dict, None)
+          Allowed IPv4 or IPv6 addresses.
 
-    type (optional, str, linode)
-      The type of device to be attached to this Firewall.
 
+              ipv4 (False, list, None)
+             A list of IPv4 addresses or networks.
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:64]: Is the indentation here correct?

File: docs/modules/firewall.rst [hunk 2]
```diff
@@ -12,124 +12,143 @@ firewall
...
 
+             Must be in IP/mask format.
 
-  rules (optional, dict, None)
-    The inbound and outbound access rules to apply to the Firewall.
 
+              ipv6 (False, list, None)
+             A list of IPv4 addresses or networks.
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:70]: Is the indentation here correct?

File: docs/modules/firewall.rst [hunk 3]
```diff
@@ -12,124 +12,143 @@ firewall
...
-      The default behavior for outbound traffic.
 
+          description (False, str, None)
+          A description for this rule.
 
-    inbound (optional, list, None)
-      A list of rules for inbound traffic.
 
+          label (True, str, None)
+          The label of this rule
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:81]: The sentence has no punctuation _(e.g. a period)_.

File: docs/modules/firewall.rst [hunk 4]
```diff
@@ -12,124 +12,143 @@ firewall
...
-        The description of this rule.
+          protocol (False, str, None)
+          The type of network traffic to allow.
 
 
-      addresses (optional, list, None)
-        Allowed IPv4 or IPv6 addresses.
 
+      inbound_policy (False, str, None)
+       The default behavior for inbound traffic.
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:96]: Is the indentation correct here?

File: docs/modules/firewall.rst [hunk 5]
```diff
@@ -12,124 +12,143 @@ firewall
...
 
-        ipv4 (optional, list, None)
-          A list of IPv4 addresses or networks.
 
-          Must be in IP/mask format.
+      outbound (False, list, None)
+       A list of rules for outbound traffic.
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:100]: Is the indentation correct here?

File: docs/modules/firewall.rst [hunk 6]
```diff
@@ -12,124 +12,143 @@ firewall
...
+          Controls whether traffic is accepted or dropped by this rule.
 
-          Must be in IP/mask format.
 
+          addresses (False, dict, None)
+          Allowed IPv4 or IPv6 addresses.
 
 
+              ipv4 (False, list, None)
+             A list of IPv4 addresses or networks.
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:112]: Is the indentation correct here?

File: docs/modules/firewall.rst [hunk 7]
```diff
@@ -12,124 +12,143 @@ firewall
...
 
-    outbound (optional, list, None)
-      A list of rules for outbound traffic.
+             Must be in IP/mask format.
 
 
-      label (True, str, None)
-        The label of this rule.
+              ipv6 (False, list, None)
+             A list of IPv4 addresses or networks.
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:118]: Is the indentation correct here?

File: docs/modules/firewall.rst [hunk 8]
```diff
@@ -12,124 +12,143 @@ firewall
...
-      description (optional, str, None)
-        The description of this rule.
+          description (False, str, None)
+          A description for this rule.
 
 
-      addresses (optional, list, None)
-        Allowed IPv4 or IPv6 addresses.
+          label (True, str, None)
+          The label of this rule
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:129]: No punctuation.

File: docs/modules/firewall.rst [hunk 9]
```diff
@@ -12,124 +12,143 @@ firewall
...
-        ipv6 (optional, list, None)
-          A list of IPv4 addresses or networks.
+          protocol (False, str, None)
+          The type of network traffic to allow.
 
-          Must be in IP/mask format.
 
 
+      outbound_policy (False, str, None)
+       The default behavior for outbound traffic.
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:144]: Is the indentation correct here?

File: docs/modules/firewall.rst [hunk 10]
```diff
@@ -12,124 +12,143 @@ firewall
...
 
 
 
+  status (False, str, None)
+    The status of this Firewall.
+
+
```
Comment (al-FJDh7) [inline docs/modules/firewall.rst:151]: Is the indentation correct here? ... also, the extra trailing newlines

**Review by al-FJDh7 — CHANGES_REQUESTED** [review]

File: docs/modules/firewall.rst (hunk 10 above)
Comment (LBGarber) [reply docs/modules/firewall.rst:151]: The trailing newlines are unfortunately unavoidable due to the way we handle templates :disappointed:

**Pull request merged**